Removal of license manager dependency

andizer opened the pull request:
## Context
<!--
What do we want to achieve with this PR? Why did we write this code?
-->

* The license manager isn't used anymore and the dependency can removed. This also mean that old code related to license manager behaviour can removed.

## Summary

<!--
Attach one of the following labels to the PR: `changelog: bugfix`, `changelog: enhancement`, `changelog: other`, `changelog: non-user-facing`.
If the changelog item is a bugfix, please use the following sentence structure: Fixes a bug where ... would ... (when ...).
If the changelog item is meant for the changelog of another repo, start you changelog item with the repo name between square brackets, for example: * [wordpress-seo-premium] Fixes a bug where ....
If the same changelog item is applicable to multiple changelogs/repos, add a separate changelog item for all of them.
-->
This PR can be summarized in the following changelog entry:

* N/A - Removal of all license manager related code.

## Relevant technical choices:

* Decided to move logic from `WPSEO_Extensions` to `WPSEO_Addon_Manager` this makes it possible to deprecate the `WPSEO_Extensions`. Used the slugs for references.
* I think we can remove the class `WPSEO_License_Page_Manager`, but maybe we want some logic that adds the notices, but this should be based on the `WPSEO_Addon_Manager`

Todo:
* [x] Check if all addons still work
* [x] Convert `WPSEO_Extension` to an array and deprecate the `WPSEO_Extension` as well. There is one place where it is used now, no need to have a value object
* [x] Deprecate `WPSEO_License_Page_Manager` or convert it to an admin integration?

## Test instructions
<!--
Please follow these guidelines when creating test instructions:
- Please provide step-by-step instructions how to reproduce the issue, if applicable.
- Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->
This PR can be tested by following these steps:

* Make sure the licenses page still works. 
  * It should show the different addons. Have WooCommerce active to see the box for Yoast SEO for Woocommerce is visible
  * Have an addon enabled to the it's state being installed.

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Documentation
* [ ] I have written documentation for this change.

## Quality assurance

* [ ] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Related https://github.com/Yoast/wordpress-seo/issues/12476

omarreiss commented on inc/class-addon-manager.php:
```diff
@@ -237,6 +238,86 @@ public function has_installed_addons() {
+		];
+
+		foreach ( $addons as $product_name => $slug ) {
+			$notification = $this->create_notification( $product_name );
```
Maybe add this within the if statement so it's only done when needed?

andizer replied:
We also need this to remove the notification which will be called when condition the if-statement isn't met.

omarreiss commented on inc/class-upgrade.php:
```diff
@@ -725,11 +726,29 @@ private function upgrade_145() {
+	/**
+	 * Performs the 15.1 upgrade
+	 */
+	private function upgrade_151() {
```
Not sure I understand what this upgrade method does. Could you please explain?

andizer replied:
I removes the notifications set when a user has an old version of the plugin. This is because the older versions calls the old edd endpoint. 

I think we should remove this logic. Do you agree?

omarreiss reviewed (changes requested):
Looking good! Three questions:
- Two small comments. Can you please process those?
- Did you also test these changes in Premium? (Sometimes behaves a little differently from add-ons).
- As I can see, you deprecated license page manager. I guess you can tick the box in your PR?

andizer commented:
@omarreiss I removed the upgrade routine since it is 1.5 year ago that we introduced the new addon manager that communicatie with my-yoast. The routine removed a couple of notifications which are set after calling the edd endpoint and when the user hasn't a valid license.

omarreiss reviewed (approved):
Looks good to me. Would like another review and test by @herregroen

omarreiss commented:
@andizer also, I didn't get an answer to one of my earlier questions:
>Did you also test these changes in Premium? (Sometimes behaves a little differently from add-ons).
Could you please confirm?

andizer commented:
@omarreiss Just tested it and it works as I would expect it should work.
